ui/js/directives/treeherder/clonejobs.js

```diff
@@ -526,29 +549,33 @@ treeherder.directive('thCloneJobs', [
 
             //Empty the job column before populating it
             jobTdEl.empty();
-
-            var jgObj, jobGroup, i;
-            for (i=0; i<jobGroups.length; i++) {
-
-                jgObj = jobGroups[i];
-
+            var jgObj, jobTdHtml = "";
+            jobGroups.forEach(function (jobGroup) {
+                jgObj = jobGroup;
                 if (jgObj.symbol !== '?') {
                     // Job group detected, add job group symbols
-                    jobGroups[i].grkey = jgObj.mapKey;
-                    jobGroups[i].collapsed = true;
-                    jobGroup = $(jobGroupInterpolator(jobGroups[i]));
-                    jobTdEl.append(jobGroup);
-
+                    jobGroup.grkey = jgObj.mapKey;
+                    jobGroup.collapsed = true;
                     if (isGroupExpanded(jgObj)) {
-                        addJobBtnEls(jgObj, jobGroup.find(".group-job-list"));
+                        jobGroup.btnHTML = getJobBtnEls(jgObj);
                     } else {
-                        addGroupJobsAndCounts(jgObj, jobGroup);
+                        var btnArrays = renderGroupJobsAndCounts(jgObj);
+                        jobGroup.btnHTML = "";
```

Comment:
I think it's better not to assign btnHTML to jobGroup, since that's going to stick around and we don't want to unnecessarily consume memory. Instead, let's just stick it and countBtnHTML in temporary variables, then create an object to pass to jobGroupInterpolator below:

```js
var btnHTML, countBtnHTML;
...
jobTdHtml += jobGroupInterpolator({
    btnHTML: btnHTML,
    countBtnHTML: countBtnHTML,
    ...
})
```